contracts/treasury/Collector.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.8.10;
+
+import {IERC20} from '@aave/core-v3/contracts/dependencies/openzeppelin/contracts/IERC20.sol';
+import {ICollector} from './interfaces/ICollector.sol';
+import {VersionedInitializable} from '@aave/core-v3/contracts/protocol/libraries/aave-upgradeability/VersionedInitializable.sol';
+
+/**
+ * @title Collector
+ * @notice Stores the fees collected by the protocol and allows the fund administrator to approve or transfer the collected ERC20 tokens.
+ * @author Aave
+ **/
+contract Collector is ICollector, VersionedInitializable {
+  event NewFundsAdmin(address indexed fundsAdmin);
+
+  address internal _fundsAdmin;
+
+  uint256 public constant REVISION = 1;
+
+  /// @inheritdoc ICollector
+  function initialize(address reserveController) external initializer {
+    _setFundsAdmin(reserveController);
+  }
+
+  modifier onlyFundsAdmin() {
```

Comment:
- missing natspec
- would move the modifier definition before any function for consistency